### PR TITLE
added automated tests

### PR DESCRIPTION
**Pull request by chris-w-mclean-shopify**
### WHY are these changes introduced?

- Added automated tests
- Turned example from string component to component to fix build error

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/InContextLearning/tests/InContextLearning.test.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+          <div>{stepTwoContent}</div>
+        </InContextLearning.Step>
+        <InContextLearning.Step stepIndex={2}>
+          <div>{stepThreeContent}</div>
+        </InContextLearning.Step>
+        <InContextLearning onDismiss={onDismiss}/>
+      </InContextLearningContextProvider>
+    );
+
+    let closeButton;
```
Comment (chris-w-mclean-shopify) [inline polaris-react/src/components/InContextLearning/tests/InContextLearning.test.tsx:43]: The reason for the button reordering is that the buttons pop in and out based on the number step you are on. We could handle this better with explicit ids or selectors on the components, but I figured that would be better off as its own PR.

**Review by rockinrobin714 — APPROVED** [review]
LGTM
